### PR TITLE
Added test cases and fixed some old ones

### PR DESCRIPTION
**Pull request by kliu513**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (rennergade) [inline .vscode/settings.json:1]: This file shouldn't be added to the commit. I thought the gitignore was ignoring these, not sure why this was able to be added!

File: tests/applications/busybox/include/linux/netfilter/xt_connmark.h [hunk 2]
```diff
@@ -1,42 +1,7 @@
-/* SPDX-License-Identifier: GPL-2.0+ WITH Linux-syscall-note */
```
Comment (rennergade) [inline tests/applications/busybox/include/linux/netfilter/xt_connmark.h:1]: also wondering how this got changed and added?

**Review by rennergade — CHANGES_REQUESTED** [review]
There are some extra files being commited, please remove those.

I think this looks pretty good otherwise. It would be nice if each Python parser had a 1-2 sentence comment at the top explaining what it's accomplishing.

File: tests/applications/busybox/include/linux/netfilter/xt_connmark.h (hunk 2 above)
Comment (kliu513) [reply tests/applications/busybox/include/linux/netfilter/xt_connmark.h:1]: I don't know. It seems that these files were changed automatically on my laptop and I did a stage-all by accident. My bad. Should I just change them back?

Comment (rennergade) [reply tests/applications/busybox/include/linux/netfilter/xt_connmark.h:1]: You should be able to just remove them from the commit. Look up how to do that with git.

**Comment by kliu513** [comment]
Have made changes to the files accordingly.

**Review by rennergade — COMMENTED** [review]
There is still a remove of this file  tests/applications/php/tests/lang/include_files/eval.inc

Can you remove that from the git commit as well?

**Comment by kliu513** [comment]
Fixed. 
These files are stubborn ones on my laptop though. These automatic changes made to them cannot be discarded, and even if I add them to .gitignore, they will be staged if I do a stage-all. I'm so confused.

**Comment by rennergade** [comment]
I think it's worthwhile to be more specific with your `git add`s. It can be annoying but saves trouble down the line.

Anyways, this looks good! Merging now. Thanks!

**Pull request merged**